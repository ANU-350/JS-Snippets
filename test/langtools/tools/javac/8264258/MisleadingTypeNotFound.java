/*
 * @test /nodynamiccopyright/
 * @bug 8264258
 * @summary unknown.Class.default gives misleading compilation error
 * @compile/fail/ref=MisleadingTypeNotFound.out -Xlint:all -Werror -XDrawDiagnostics -XDdev MisleadingTypeNotFound.java
 */
package knownpkg;

public class MisleadingTypeNotFound {

    void classNotFound() {
        // Not found, but in an existing package
        Class<?> c1 = knownpkg.NotFound.class;

        // Not found, but in a (system) package which exists and is in scope
        Class<?> c2 = java.lang.NotFound.class;

        // Not found, on a genuinely unknown package
        Class<?> c3 = unknownpkg.NotFound.class;

        // Not found, but in the 'java' package which is in scope as per JLS 6.3 and observable as per JLS 7.4.3
        Class<?> c4 = java.NotFound.class;
    }
}
