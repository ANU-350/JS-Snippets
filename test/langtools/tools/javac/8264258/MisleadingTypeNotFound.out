MisleadingTypeNotFound.java:13:31: compiler.err.cant.resolve.location: kindname.class, NotFound, , , (compiler.misc.location: kindname.package, knownpkg, null)
MisleadingTypeNotFound.java:16:32: compiler.err.cant.resolve.location: kindname.class, NotFound, , , (compiler.misc.location: kindname.package, java.lang, null)
MisleadingTypeNotFound.java:19:33: compiler.err.doesnt.exist: unknownpkg
MisleadingTypeNotFound.java:22:27: compiler.err.cant.resolve.location: kindname.class, NotFound, , , (compiler.misc.location: kindname.package, java, null)
4 errors
